server/src/main/java/org/opensearch/action/admin/indices/replication/TransportSegmentReplicationStatsAction.java

```diff
@@ -134,6 +143,13 @@ protected SegmentReplicationStatsResponse newResponse(
                 }
             }
         }
+        // combine the search replica stats with the stats of other replicas
+        for (Map.Entry<String, List<SegmentReplicationPerGroupStats>> entry : primaryStats.entrySet()) {
```

Comment:
Sorry didn't catch this on first pass, but looks like all the stats are computed from the state object returned by the replica.  Can we not return that here similar to normal replicas and compute the stats here?  Then you don't need to make updates to the transport object & it behaves just like the other replicas

Reply:
Yes. We need `Completed` and `OngoingState` to compute search replica stats which we can get from the `targetService` if we have the `shardId`(which we can get from `shardRouting`). So we can compute this in coordinator.